pkg/cloud/services/compute/instance.go

```diff
@@ -276,6 +276,9 @@ func (s *Service) createInstance(eventObject runtime.Object, clusterName string,
 			}
 			return false, err
 		}
+		if createdInstance.State == infrav1.InstanceStateError {
+			return false, fmt.Errorf("openstack instance %s go error.", createdInstance.ID)
```

Comment:
```suggestion
			return false, fmt.Errorf("error creating OpenStack instance %s, status changed to error", createdInstance.ID)
```

Maybe a bit more convincing error message